bin/generate.py

```diff
@@ -1023,7 +1033,7 @@ def needed(
                         if not t.config.solution.generate_interaction(bar, cwd, t):
                             return False
                         used_solution = True
-                        meta_yaml["interactor_hash"] = interactor_hash
+                        meta_yaml.interactor_hash = cast(dict[Any, Any], interactor_hash)
```

Comment:
Shouldn't the type of `MetaYaml.interactor_hash` be `dict[str, dict[str, str]]` instead of `dict[object, object]`? That would avoid the need for a `cast` due to the invariance of key/value types in `dict` :slightly_smiling_face:

Reply:
That is true from a logical perspective, but never verified. If we define it as `dict[str, dict[str, str]]` we would need to actually verify this type while parsing the MetaYaml (both keys and values could be something weird). But for the current code this actually never needed. Thus, I would argue that it's cleaner to assume the most general type which avoids any need for type checking during the read of the yaml file. That is also always safe, since by definition any key and any value is an object... MyPy just doesn't understand this ):